app/src/main/java/com/gut/follower/activities/SplashActivity.java

```diff
@@ -3,16 +3,23 @@
 import android.content.Intent;
 import android.os.Bundle;
 import android.support.annotation.Nullable;
-import android.support.v7.app.AppCompatActivity;
 
-public class SplashActivity extends AppCompatActivity{
+import com.gut.follower.utility.SessionManager;
+
+public class SplashActivity extends BaseActivity{
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        Intent intent = new Intent(this, LoginActivity.class);
-        startActivity(intent);
-        finish();
+        if(SessionManager.isUserLoggedIn(getApplicationContext())){
+            Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+            startActivity(intent);
+        } else {
+            Intent intent = new Intent(getApplicationContext(), LoginActivity.class);
+            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+            startActivity(intent);
+        }
```

Comment:
You repeat yourself here and in some other places. Try to consider excluding this three lines of code (maybe even to BaseActivity) as a method and send activity type as an argument.